### ISSUE TITLE
Media playback broken on certain non-youtube websites

### ISSUE THREAD

**Issue by abogatskiy** [AdguardTeam/CoreLibs#1944]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

2.16.0.1857

### Browser version

Safari 18.2

### OS version

Sequoia 15.2

### Which DNS server do you use?

DNS protection disabled

### DNS protocol

None

### Issue Details

Steps to reproduce:
1. Enable AdGuard for Mac
2. Disable all filters, DNS protection, stealth mode, all extensions. The only enabled feature in the settings is "Automatically filter applications" (specifically Safari)
3. Disable the AdGuard Assistant extension in Safari.
4. Go to certain pages containing media, for example https://archive.org/details/the-spanish-civil-war-BBC/THE+SPANISH+CIVIL+WAR+-+Episode+1_+Prelude+To+Tragedy+(HISTORY+DOCUMENTARY)+(360p_25fps_H264-128kbit_AAC).mp4 
5. Another example is the audio preview here https://libro.fm/audiobooks/9781526621757-the-international-brigades?srsltid=AfmBOoqPrUlbS3y3M4rS8DuOKX7iNxFHU0CgiiOEAzMdh5BFNQMqVPeV
6. Click the play button.

### Expected Behavior

Video/audio starts playing

### Actual Behavior

The video never starts playing, eventually spits out an error.

### Screenshots

<details><summary>Playback error</summary>

<img width="1160" alt="Screenshot 2024-12-18 at 10 39 18 AM" src="https://github.com/user-attachments/assets/74a7b43a-d8a1-46c0-8e7c-b0f62fefdd30" />
</details>

<details><summary>The only checkmarked settings items</summary>

<img width="812" alt="Screenshot 2024-12-18 at 10 44 09 AM" src="https://github.com/user-attachments/assets/f79139e7-3353-4af4-a8a3-e06e00c2d09d" />

</details>

### Additional Information

This is a weird issue that seems to occur even when all actual protection in AdGuard is turned off. I've tested it on two different Macs and it's the same on both. I tested an independent network connection as well.

The libfro.fm audio can be played, but only after reloading the page once. On archive.org reloading isn't sufficient. It's not completely clear which pages/websites cause the issue. Other videos on archive.org or previews on libro.fm appear to play fine. However, overall it seems like this is affecting a huge portion of non-youtube media.

The only way to circumvent the issue that I've found is to use AdGuard's menu bar toggle to turn it off completely, or to uncheck "Automatically filter applications". I've also tried disabling all other Safari extensions, to no avail. The issue seems to have started relatively recently, perhaps with Sequoia 15.1 or so.

The issue persists with filters turned on. Meanwhile using another browser (Firefox) fixes the issue — even with filtering turned on. With AdGuard for iOS there is also no issue (including with filtering turned on).

**Comment by AlexandrPkhm** [comment]
@abogatskiy Hello!

We are investigating this issue. For now, as a workaround, we recommend that you open **AdGuard** → **Gear** icon → **Advanced** → **Advanced settings...** then find `network.extension.http.downgrade.bundleids` and add `com.apple.Safari` to this field.

**Comment by abogatskiy** [comment]
That worked, thanks a lot for the quick workaround!

**Comment by sfionov** [comment]
Adguard for Mac 2.16.2 is released with the proper fix, the workaround is not needed anymore.

**Comment by cassperine** [comment]
here the 2.16.2 doesn't stop crashing on macOS 15.2 😭 The previous version was fine
<img width="565" alt="Screenshot 2024-12-24 at 12 16 01" src="https://github.com/user-attachments/assets/5d85d8ea-9fe1-4cda-be32-e043844982c0" />

**Comment by sfionov** [comment]
@cassperine Can you please provide steps to reproduce?

**Comment by sfionov** [comment]
@cassperine Or can you describe circumstances under which this is happened? 

It would be also useful if you collect debug logs to understand better what went wrong.
Here's what is needed to collect them:
1. Click **AdGuard icon** in the menu bar *→ Gear → Advanced  → Logging → Logging level → Debug*.
2. Reproduce the problem, then remember the exact time it happened.
3. *Menu → Advanced → Logging → Export Logs and System Info.…*.
4. Send this file to [qa@adguard.com](mailto:qa@adguard.com):
   — include `[mac]` keyword and `ISSUE_NUMBER` in the subject of your email
   — specify the exact time when the issue occurred

**Comment by cassperine** [comment]
@sfionov I sent the email with log! When I'm browsing in Safari the application crashes, especially on YouTube and Reddit. In Chrome there is no problem.

**Comment by sfionov** [comment]
@cassperine Thanks a lot, created the separate issue #1945. The workaround is to temporarily turn off HTTP/3 filtering in the Advanced settings.
